xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/macros.vm

```diff
@@ -387,18 +387,20 @@ $xwiki.getUserName("xwiki:${username}")
   #end
   #if(!$picturelist.containsKey("${username}--${size}"))
     #getUserAvatarURL($username $avatarURL $size)
+    #set ($resolvedUser = $services.model.resolveDocument($username))
+    #set ($userNameEscaped = $escapetool.html($xwiki.getPlainUserName($resolvedUser)))
```

Comment:
I feel you should use getUserName instead of escaping the result of getPlainUserName since getUserName job is to return html content already.